daemon/list_test.go

```diff
@@ -140,3 +189,30 @@ func TestNameFilter(t *testing.T) {
 	assert.Assert(t, is.Len(containerListWithPrefix, 1))
 	assert.Assert(t, containerListContainsName(containerListWithPrefix, three.Name))
 }
+
+func TestLimitFilter(t *testing.T) {
+	db, err := container.NewViewDB()
+	assert.NilError(t, err)
+	d := &Daemon{
+		containersReplica: db,
+	}
+
+	// start a number of containers
+	num := rand.Intn(64)
+	containers := make([]*container.Container, num)
+	for i := range num {
+		name := fmt.Sprintf("cont-%d", i)
+		containers[i] = setupContainerWithName(t, name, d)
+	}
+
+	// list them with the limit option and verify correctness; note that the limit
+	// option only limits the size of the list, but there are no guarantees as to
+	// which containers are in that list (e.g., if limit is 10, it does not mean
+	// the list will have the first 10 containers we created; it will have any 10
+	// of the containers we created).
+	limit := rand.Intn(64)
+	containerList, err := d.Containers(context.Background(), &containertypes.ListOptions{Limit: limit})
+	assert.NilError(t, err)
+	expectedListLen := int(math.Min(float64(num), float64(limit)))
```

Comment:
Can use the builtin `min` (and `max`) now; no need to use the math package 😄 

```suggestion
	expectedListLen := min(num, limit)
```

Reply:
Nice, I'll change it!

Reply:
Done.